### PR TITLE
Integration of double-batched FFT library (bbfft)

### PR DESCRIPTION
**Pull request by omarkahmed**
This is an updated PR accounting to clean up the git history.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,16 @@ option(YAKL_B4B "Force serial computation (default OFF)." OFF)
...
 
+
+set(YAKL_SYCL_BBFFT                  OFF                  CACHE BOOL    "Use Double-Batched FFT")
+set(YAKL_SYCL_BBFFT_AOT              OFF                  CACHE BOOL    "Use Double-Batched FFT AOT")
+set(YAKL_SYCL_BBFFT_HOME             $ENV{BBFFT_HOME}     CACHE PATH    "Double-Batched FFT Home directory")
+set(YAKL_SYCL_BBFFT_AOT_LEGACY_UMD   OFF                  CACHE BOOL    "Use older PVC user mode driver")
+
+set(YAKL_CXX_SOURCE src/YAKL_fortran_interfaces.cpp)
+set(YAKL_F90_SOURCE src/YAKL_gator_mod.F90)
+set(YAKL_SOURCE ${YAKL_CXX_SOURCE} ${YAKL_F90_SOURCE})
```
Comment (mrnorman) [inline CMakeLists.txt:37]: These lines need to be removed.

**Comment by mrnorman** [comment]
After going through this, I have a few thoughts to start with:
* The new files in the FFT unit test should be prefixed with `bbfft_` or moved to a `bbfft` folder
* I'm concerned about the amount of CMake code used in the FFT unit test CMake file. Is there a way we can pre-compile bbfft and not have to have all of the CMake workflow? YAKL users aren't going to want to have to add that sort of thing to their codes. Maybe we can talk more about this on the next Intel-E3SM concall.

File: unit/FFT/generate.cpp
[the remark this answers is not in the document]
Comment (omarkahmed) [reply unit/FFT/generate.cpp:1]: Yes, that's correct.  And AOT mode, specifically.

**Comment by omarkahmed** [comment]
@mrnorman , thanks for the feedback.

The extra CMake syntax (and source code) in the FFT unit test is specific to the AOT compilation pathway, so is not necessary if the user wants to compile JIT (which unfortunately eliminates much of the value of bbfft).  Will do some thinking to see how we can reduce (hopefully elimiante) that complexity, and we can discuss next week.

**Comment by omarkahmed** [comment]
@mrnorman & @abagusetty , the extra cmake complexity is related to the AOT implementation for bbfft, where the FFT sizes are specified at compile time: https://github.com/intel/double-batched-fft-library/blob/develop/examples/aot/CMakeLists.txt.  The older implementation in YAKL had this complexity in the main YAKL cmake file as opposed to the unit tests (or the YAKL user code), but I moved that to the unit test in order to preserve YAKL as a header-only library: https://github.com/omarkahmed/YAKL/commit/d6ecec3600360b7eb5b80a3a9321fb254c1d9069

We can build a custom version of bbFFT that compiles in some standard FFT sizes, and deploy that on JLSE and Sunspot.  But wondering if that might be an issue in terms of portability from the E3SM team perspective?

**Pull request closed without merging**